packages/metadata_attacher/src/index.ts

```diff
@@ -101,16 +141,140 @@ const getEmbeddedMetadataFromS3 = async (
 		metadata,
 	);
 
-	const objectCharacteristicsExtensionSection =
-		administrativeSection["mets:techMD"]["mets:mdWrap"]["mets:xmlData"][
-			"premis:object"
-		]["premis:objectCharacteristics"]["premis:objectCharacteristicsExtension"];
+	const {
+		"mets:techMD": {
+			"mets:mdWrap": {
+				"mets:xmlData": {
+					"premis:object": {
+						"premis:objectCharacteristics": {
+							"premis:objectCharacteristicsExtension":
+								objectCharacteristicsExtensionSection,
+						},
+					},
+				},
+			},
+		},
+	} = administrativeSection;
+
+	return objectCharacteristicsExtensionSection;
+};
+
+const getPhotoMetadata = (
+	embeddedMetadata: EmbeddedMetadata,
+): {
+	preciseTimestamp: Date | undefined;
+	edtfTime: string | undefined;
+	tags: string[] | undefined;
+	description: string | undefined;
+	title: string | undefined;
+	altText: string | undefined;
+} => {
+	const {
+		"rdf:RDF": { "rdf:Description": rdfMetadata },
+	} = embeddedMetadata;
+
+	const timestamp =
+		rdfMetadata["ExifIFD:DateTimeOriginal"] === undefined
+			? undefined
+			: getISOStringFromExifTimestamp(
+					rdfMetadata["ExifIFD:DateTimeOriginal"],
+					rdfMetadata["ExifIFD:OffsetTimeOriginal"],
+				);
+	const tags =
+		rdfMetadata["IPTC:Keywords"] === undefined
+			? undefined
+			: rdfMetadata["IPTC:Keywords"]["rdf:Bag"]["rdf:li"];
+
+	return {
+		preciseTimestamp:
+			rdfMetadata["ExifIFD:OffsetTimeOriginal"] === undefined ||
+			timestamp === undefined
+				? undefined
+				: new Date(timestamp),
+		edtfTime: timestamp,
+		tags,
+		description:
+			rdfMetadata["IPTC:Caption-Abstract"] ??
+			rdfMetadata["ExifIFD:UserComment"] ??
+			rdfMetadata["ExifIFD:Comments"],
+		title: rdfMetadata["IPTC:ObjectName"] ?? rdfMetadata["ExifIFD:Title"],
+		altText: rdfMetadata["XMP-iptcCore:AltTextAccessibility"],
+	};
+};
+
+const getVideoMetadataFromMediaInfo = (
+	embeddedMetadata: EmbeddedMetadata,
+):
+	| {
+			preciseTimestamp: Date | undefined;
+			edtfTime: string | undefined;
+	  }
+	| undefined => {
+	const tracks = embeddedMetadata.MediaInfo?.media.track;
+	const generalTracks = tracks?.filter(
+		(track: TrackMetadata) => track.StreamKind === "General",
+	);
+	if (generalTracks?.[0] === undefined) {
+		return undefined;
+	}
+	if (generalTracks.length > 1) {
```

Comment:
I think we could improve legibility of this function a bit with some minor tweaks:

(1) Check if tracks is undefined before the filter happens, if so, boop out.  By doing this we have guarded against both `tracks` and `generalTracks` being possibly undefined, and can ditch the `?`.

(2) Maintain this defensive check to ensure we don't have `generalTracks.length > 1`.

(3) Extract `const [generalTrack] = generalTracks` (removing the need for index access later)

(4) Then wrap up with the above check that the generalTrack is not undefined, but now as `generalTrack === undefined => boop out` which is a bit more self documenting.